File: Week2/Day13/bj2798_rank.cs
// https://www.acmicpc.net/source/24531694
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string[] a = Console.ReadLine().Split();
        int num = int.Parse( a[0]);
        int sum = int.Parse(a[1]);
        string[] aa = Console.ReadLine().Split();
        int[] value = new int[aa.Length];
        for (int i = 0; i < aa.Length; i++)
        {
            value[i] = int.Parse(aa[i]);
        }
        int result = new int();
        for (int i = 0; i < num-2; i++)
        {
            for (int k = i+1; k < num-1; k++)
            {
                for (int j = k+1; j < num; j++)
                {
                    // 1 2 3 4 5
                    int sum2 = value[i] + value[k] + value[j];
                    if (sum2 <= sum&& sum-result > sum-sum2)
                    {
                        result = sum2;
                    }
                }
            }
        }
        Console.WriteLine(result);
    }
 
}

File: Week2/Day13/bj2798_rank.java
// https://www.acmicpc.net/source/23868424
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.stream.IntStream;
import java.io.IOException;
 
public class Main {
	
	public static StringBuilder sb = new StringBuilder();
 
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer str = new StringTokenizer(br.readLine(), " ");
		
		int num = Integer.parseInt(str.nextToken());
		int target = Integer.parseInt(str.nextToken());
		
		str = new StringTokenizer(br.readLine(), " ");
		int[] card = new int[num];
		for (int i = 0; i < num; i++) 
			card[i] = Integer.parseInt(str.nextToken());
		
		
		System.out.println(find(card, num, target));
		 
	}
	
	public static int find (int[] card, int num, int target) {
		int answer = 0;
		for (int i = 0; i < num-2; i++) {
			for (int j = i + 1; j < num-1; j++) {
				for (int k = j + 1; k < num; k++) {
					int sum = card[i]+card[j]+card[k];
					if (sum == target) return sum;
					if (answer < sum && sum < target) answer = sum;
				}
			}
		}
		return answer;
	}
}